### PR TITLE
Tag logger field with the PID of a Command's exec process

### PR DESCRIPTION
**Pull request by cheapRoc**
Ref: #497 

This adds a Command's process PID as a logger field which displays the PID as a tag in the logs. To implement this I've added a new attribute on the Command struct called `fields` which passes around the state of the logger fields in case they need to be reset (as we do here).

### CODE REVIEW

File: commands/commands.go [hunk 1]
```diff
@@ -99,6 +101,19 @@ func (c *Command) Run(pctx context.Context, bus *events.EventBus) {
...
+		// if we're able to, log the PID of our Command's exec process through
+		// our logger fields
+		if c.Cmd != nil && c.Cmd.Process != nil {
+			pid := c.Cmd.Process.Pid
+			if len(c.fields) > 0 {
+				c.fields["pid"] = pid
+			} else {
+				c.fields = log.Fields{"pid": pid}
+			}
+			c.logger = *log.WithFields(c.fields)
```
Comment (tgross) [inline commands/commands.go]: I'm pretty sure this breaks the "raw" logging, doesn't it? I think we only want to assign to `c.logger` in the case where `c.fields > 0`, so this line should be in first branch of the `if` above.

Comment (tgross) [inline commands/commands.go]: Actually, we attach to the process stdout/stderr well before this part -- so does this work? We should definitely verify the pid field shows up in the `test_logging` integration test.

Comment (cheapRoc) [inline commands/commands.go]: You're right, and I completely missed the comment about that.

**Comment by cheapRoc** [comment]
This will be reviewed and merged after I close #476 (or #499) since there was a race discovered that became more prominent after this was added (#500).

**Comment by cheapRoc** [comment]
Since I haven't had the time to complete it's dependencies I'm going to close this for the moment and revisit later.

**Comment by cheapRoc** [comment]
This is next up in line now that #512 has been merged. I'll have a PR open soon.

**Pull request closed without merging**